src/plugins/intel_gpu/src/graph/impls/ocl_v2/sdpa/sdpa_opt.cpp

```diff
@@ -75,29 +73,23 @@ class SDPAOptImpl : public SDPAImplBase {
                     add_stage(indirect_multi_tokens, params);
 #ifdef ENABLE_ONEDNN_FOR_GPU
                 } else if (SDPAOpt::supports_micro_sdpa(params)) {
-                    GPU_DEBUG_TRACE_DETAIL << "add stage for micro_sdpa  non-dynamic with prefill_stage \n";
+                    GPU_DEBUG_TRACE_DETAIL << "add stage for micro_sdpa non-dynamic with prefill_stage \n";
                     add_stage(regular_micro_multi_tokens, params);
-                    add_stage(regular_micro_single_token, params);
+                    if (!has_stage(regular_micro_multi_tokens)) {
+                        GPU_DEBUG_TRACE_DETAIL << "fail to create micro kernel, fallback to regular_multi_tokens for prefill \n";
+                        add_stage(regular_multi_tokens, params);
+                    }
 #endif
                 } else {
                     GPU_DEBUG_TRACE_DETAIL << "add stage regular_multi_tokens kernels with prefill_stage \n";
                     add_stage(regular_multi_tokens, params);
                 }
             } else {
                 GPU_DEBUG_TRACE_DETAIL << "add stage single_tokens \n";
-                const auto& gfx_ver = params.get_program().get_engine().get_device_info().gfx_ver;
-                bool is_ARL_H = (gfx_ver.major == 12 && gfx_ver.minor == 74);
-                if (!SDPAOpt::supports_micro_sdpa(params) || is_ARL_H) {
-                    add_stage(is_indirect ? indirect_single_token : regular_single_token, params);
```

Comment:
May I know if it will remove ARL-H SDPA WA for 2nd token generation:  https://github.com/openvinotoolkit/openvino/pull/27387? Or this condition will not hit after OV migrate to PA as default backend. Thanks!